arduino-ide-extension/src/common/protocol/sketches-service.ts

```diff
@@ -1,5 +1,21 @@
+import { ApplicationError } from '@theia/core/lib/common/application-error';
 import URI from '@theia/core/lib/common/uri';
 
+export namespace SketchesError {
+  export const Codes = {
+    NotFound: 5001,
+  };
+  export const NotFound = ApplicationError.declare(
+    Codes.NotFound,
+    (message: string, uri: string) => {
+      return {
+        message,
+        data: { uri },
```

Comment:
Consider including the file-system path? Would be obsolete with the next Theia version.

Reply:
350c04b0bc28bc57acb12d65c6b12cff1777ccea + https://github.com/arduino/arduino-cli/issues/1797